Added missing docs for the `wrapperProps` property for the `BlockListBlock` component returned by the `editor.blockListEdit` filter.

hansjovis opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
<!-- Please describe what you have changed or added -->
Fixes https://github.com/WordPress/gutenberg/issues/24864

Adds missing documentation on how to use the `wrapperProps` property on the `BlockListBlock` component to add (custom) properties to a block's wrapper component, using the `editor.blockListEdit` filter.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
I tested the added code samples by creating a minimal plugin that extended the block editor using the steps [as described in the Gutenberg Handbook](https://developer.wordpress.org/block-editor/tutorials/javascript/js-build-setup/). Once using the ES5 syntax, once using the ESNext syntax.

## Screenshots <!-- if applicable -->
N/A

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Added missing documentation.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

talldan commented on docs/designers-developers/developers/filters/block-filters.md:
```diff
@@ -319,6 +319,51 @@ wp.hooks.addFilter( 'editor.BlockListBlock', 'my-plugin/with-client-id-class-nam
+
+const withClientIdClassName = createHigherOrderComponent( ( BlockListBlock ) => {
+	return ( props ) => {
+		const wrapperProps = { 'data-your-property': 'your-value' };
+		return <BlockListBlock { ...props } wrapperProps={ wrapperProps } />;
```
Thanks for contributing to the docs!

Seems like this would overwrite existing `wrapperProps`, could it be changed to something like:
```suggestion
		const wrapperProps = { 
		    ...props.wrapperProps, 
		    'data-your-property': 'your-value',
		};
		return <BlockListBlock { ...props } wrapperProps={ wrapperProps } />;
```

The ES5 example has the same issue

mkaz committed:
Update wrapperProps docs to new documentation file

- The docs were reorganized so rebasing required moving the changes in
  the original PR

- Updated the function name to match the wrapperProp isntead of clientId
  example from previous

 - Applied same expansion change that @talldan contributed to ES5
   example

Co-authored-by: Daniel Richards <daniel.p.richards@gmail.com>

mkaz reviewed (approved):
I had to move the changes to a new file, everything looks good the only real change I made was to the function name and added the same spread expansion to the props in ES5 example that was done in ESNext example.

I'll give it a day to see if you have any tweaks, but it was mostly there, so I'll merge it in tomorrow if no issues. 👍
